lib/generators/json_serializable_generator.dart

```diff
@@ -192,6 +216,24 @@ bool _hasFromJsonCtor(DartType type) {
   return false;
 }
 
+DartType _listTypeWithFromJsonCtor(DartType type) {
+  if (type is! InterfaceType) return null;
+  if (!_isDartList(type)) return null;
+
+  var listType = (type as InterfaceType).typeArguments.first;
+  if (_hasFromJsonCtor(listType)) {
+    return listType;
+  } else {
+    return null;
+  }
+}
+
+bool _isDartList(DartType type) {
+  return type.element.library != null &&
+      type.element.library.isDartCore &&
+      type.name == 'List';
```

Comment:
might be nice to support all subtypes of Iterable but I think that is more complicated